src/sidebar/oauth-auth.js

```diff
@@ -79,14 +80,20 @@ function auth($http, settings) {
 
   // Set a timeout to refresh the access token a few minutes before it expires.
   function refreshAccessTokenBeforeItExpires(tokenInfo) {
-    var delay = tokenInfo.expiresIn * 1000;
-
     // We actually have to refresh the access token _before_ it expires.
     // If the access token expires in one hour, this should refresh it in
     // about 55 mins.
-    delay = Math.floor(delay * 0.91);
+    var refreshAfter = tokenInfo.expiresAt.valueOf() * 0.91;
```

Comment:
This computation is wrong and causing it to refresh too often, I'm writing a new one